colcon_ros/task/cmake/build.py

```diff
@@ -36,6 +37,12 @@ async def build(self):  # noqa: D102
         additional_hooks = create_pkg_config_path_environment_hooks(
             Path(args.install_base), self.context.pkg.name)
 
+        # register additional custom hooks
+        custom_hooks_path = Path(args.install_base) / \
+            'share' / self.context.pkg.name / 'env_hook'
```

Comment:
Is this known location documented somewhere? Should it be configurable?
I ask because even within the ROS 2 project I've noticed differences in the name of the directory (e.g. "env-hooks" vs. "env_hooks").

Reply:
> Is this known location documented somewhere? 

I will document it once it is merged.

> Should it be configurable?

How could this be configured by a package? I don't see a way to do so.

> I ask because even within the ROS 2 project I've noticed differences in the name of the director

ROS packages using either `catkin` or `ament_cmake` have their own API to register env hooks from within the CMake code of the package.

Reply:
> How could this be configured by a package?

`colcon.pkg` seems like convenient place to let the user control the directory where there hooks are installed (e.g. "hooks_dir"). Note, I don't think this needs to be done now, but might be a good todo for later.

I might not have enough context for this change; could the user instead add a `colcon.pkg` with a "hooks" entry to accomplish a similar thing as this PR (without being limited to the hard-coded install directory)?

Reply:
> I might not have enough context for this change; could the user instead add a `colcon.pkg` with a "hooks" entry to accomplish a similar thing as this PR (without being limited to the hard-coded install directory)?

:disappointed: Doh, yes, absolutely. I didn't even remember that feature existed. And using that is certainly much cleaner than this set of pull requests. I wish I would have remembered it before spending the time on implementing these changes.